src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/DetectPreviewFeatureAnalyzer.cs

```diff
@@ -942,7 +942,7 @@ private static ISymbol SymbolFromArrayCreationOperation(IArrayCreationOperation
         {
             if (!requiresPreviewFeaturesSymbols.TryGetValue(symbol, out (bool isPreview, string? message, string? url) existing))
             {
-                Debug.Assert(true, $"Should never reach this line. This means the symbol {symbol.Name} was not processed in this analyzer");
+                Debug.Assert(false, $"Should never reach this line. This means the symbol {symbol.Name} was not processed in this analyzer");
```

Comment:
This looks like a typo. Previously, the assert will always pass and we wouldn't know there is an issue.

Reply:
Yep looks weird. I would replace `Debug.Assert(false` by `Debug.Fail` which is clearer.